import-automation/workflow/ingestion-helper/import_utils.py

```diff
@@ -0,0 +1,149 @@
+"""Utility functions for the ingestion helper."""
+
+import logging
+from croniter import croniter
+from datetime import datetime, timezone
+import base64
+from googleapiclient.discovery import build
+import json
+from googleapiclient.errors import HttpError
+from google.oauth2 import id_token
+from google.auth.transport import requests
+from google.auth import jwt
+
+
+def get_caller_identity(request):
+    """Extracts the caller's email from the Authorization header (JWT)."""
+    auth_header = request.headers.get('Authorization')
+    if auth_header:
+        parts = auth_header.split()
+        if len(parts) == 2 and parts[0].lower() == 'bearer':
+            token = parts[1]
+            unverified_claims = {}
+            try:
+                unverified_claims = jwt.decode(token, verify=False)
+                #logging.info(f"Token claims (unverified): iss={unverified_claims.get('iss')}, aud={unverified_claims.get('aud')}, email={unverified_claims.get('email')}")
+                logging.warning(
```

Comment:
this log should be in the exception block or within `if not unverified_claims:`?